server/src/main/java/com/linecorp/centraldogma/server/internal/storage/project/SafeProjectManager.java

```diff
@@ -49,19 +51,19 @@ public void close(Supplier<CentralDogmaException> failureCauseSupplier) {
 
     @Override
     public boolean exists(String name) {
-        validateProjectName(name);
+        validateProjectName(name, true);
         return delegate().exists(name);
     }
 
     @Override
     public Project get(String name) {
-        validateProjectName(name);
+        validateProjectName(name, true);
```

Comment:
In short, if a plugin (i.e. ControlPlanePlugin) wants to modify an internal `dogma` project, do we need to set a `ServiceRequestContext` before calling methods?

I'm wondering if it's easier to...
- Just validate the (dogma && admin) condition at the API level (we already seem to be doing this via. `RequiresRole`, `RequiresAdministrator`, `RequiresPermission` decorators)
- Hard code disallowing deleting/purging the internal `dogma` project at this layer

Reply:
> In short, if a plugin (i.e. ControlPlanePlugin) wants to modify an internal dogma project, do we need to set a ServiceRequestContext before calling methods?

Not exactly. 😄 The plugin will use `ProjectManager` instead of this `SafeProjectManager` so a `RequestContext` isn't needed.
This change is for REST API which is used from UI. With this feature, we can easily check the repository through UI if the configuration is stored as expected. Let me add this to the commit description. 

 > we already seem to be doing this via. RequiresRole, RequiresAdministrator, RequiresPermission decorators
 
 We can do that but I'm not sure if it makes it simpler. For list() method, it includes the dogma project if it's called by an admin:
 ```java
public Map<String, Project> list() {
    final Map<String, Project> list = delegate().list();
    final Map<String, Project> ret = new LinkedHashMap<>(list.size());
    for (Map.Entry<String, Project> entry : list.entrySet()) {
        if (isValidProjectName(entry.getValue().name(), true)) {
            ret.put(entry.getKey(), entry.getValue());
        }
    }
    return Collections.unmodifiableMap(ret);
}
 ```
Doing this with the decorators, isn't simple I think. 🤔

Reply:
> The plugin will use ProjectManager instead of this SafeProjectManager so a RequestContext isn't needed.

I see, I didn't know that 👍 

> We can do that but I'm not sure if it makes it simpler.

I see, the concern I had with the current approach is:
- Admin permission is checked at multiple places, making it complex and difficult to maintain
- We need to remember to add `validateProjectName` for every API we add in the future

Having said this, if you believe the current approach is most efficient I trust your judgement

Reply:
Ah ha, I get your point. Let me encapsulate it by putting the logic that validates the dogma project in the `SafeProjectManager`.

Reply:
I've renamed `SafeProjectManager` to `ProjectApiManager` and gathered the validating logic in it. It's now also used in the Thrift service as well. PTAL. 🙇